src/CodeGen_C.cpp

```diff
@@ -1235,9 +1250,19 @@ void CodeGen_C::visit(const Call *op) {
             rhs << "_ucon, ";
         }
 
+        // TODO: special-case for https://github.com/halide/Halide/issues/2099;
+        // embedded buffers will generate code for bounds query, which can
+        // never execute (_halide_buffer_is_bounds_query() always returns false
+        // for them), but the generated bounds-query code won't compile due to 
+        // const correctness. We add some const_casts here as a temporary
+        // workaround.
```

Comment:
OK, good point. I'll revise it to be:
-- const for host data (except dimensions==0)
-- const for shape
-- non-const for buffer itself